src/ast/mod.rs

```diff
@@ -331,14 +322,10 @@ pub enum Expr {
     /// A constant of form `<data_type> 'value'`.
     /// This can represent ANSI SQL `DATE`, `TIME`, and `TIMESTAMP` literals (such as `DATE '2020-01-01'`),
     /// as well as constants of other types (a non-standard PostgreSQL extension).
-    TypedString {
-        data_type: DataType,
-        value: String,
-    },
-    MapAccess {
-        column: Box<Expr>,
-        keys: Vec<Expr>,
-    },
+    TypedString { data_type: DataType, value: String },
+    /// Access a map-like object by field (e.g. `column['field']` or `column[4]`
+    /// <https://clickhouse.com/docs/en/sql-reference/data-types/map/>
```

Comment:
Maybe it's a good idea to mention that ArrayIndex replaces MapAccess for some dialects.